webui/react/src/hooks/useSettings.ts

```diff
@@ -230,7 +272,7 @@ const useSettings = <T>(config: SettingsConfig, options?: SettingsHookOptions):
     userSettings,
```

Comment:
I have a lot of questions about the basics of this hook. For example why does `useSettings` listen to changes in the location? Shouldn't those modifications be made _by_ `useSettings`? When is that trigger happening?

(This is kind of why I was recommending rewriting it from scratch)

Reply:
for that in particular, the concept is that if you have a url that stores the current page state, you can share it and someone else can view the page in the same state

Reply:
>we don't need to listen for that.

in that case what is the mechanism for going from url params -> state.

right now we are using `location.search`

Reply:
I think that' what `useLocation` does?

However you get the info, if you use it in a hook, react is going to want you to put it in the dependency array, no?

Reply:
@trentwatt `useLocation` subscribes the component to changes in the location. `window.location` is just the raw `Location`. If we don't need to subscribe the component to changes in the location then we should just get it directly.

Reply:
are you telling me we don't have to use exhaustive deps 😃

Reply:
If you list all of the dependencies it's equivalent to not including a dependency array. The whole _purpose_ of the dependency array is to narrow the dependencies.